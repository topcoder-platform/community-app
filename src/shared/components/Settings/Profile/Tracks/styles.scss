@import "~styles/mixins";

.track-toggle {
  border-top: 1px solid $tc-gray-20;

  hr {
    margin-top: 15px;
    width: 100%;

    &.first {
      margin-top: 0;
    }

    &.bottom {
      margin-bottom: 0;
    }
  }

  .track {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: left;

    .content {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      padding: 15px 0;
      border-bottom: 1px solid $tc-gray-20;
      width: 100%;
      transition: background-color 0.15s;

      &.disabled {
        background-color: $tc-gray-neutral-light;
      }

      .track-details {
        display: flex;
        flex-direction: row;
        align-items: center;
        padding-left: 10px;

        .icon {
          &.disabled {
            color: #b7b7b7;
          }

          svg,
          img {
            height: 32px;
            width: 32px;
          }

          span {
            @include sofia-pro-medium;

            font-size: 16px;
          }
        }
      }
    }

    .text {
      margin-left: 15px;

      .title {
        font-size: 16px;
        line-height: 28px;

        @include sofia-pro-medium;

        text-transform: uppercase;
        transition: 0.1s color;

        &.disabled {
          color: #b7b7b7;
        }
      }

      .description {
        @include merriweather-sans-regular;

        font-size: 13px;
        margin-top: 4px;
        color: $tc-gray-40;
      }
    }
  }

  @media only screen and (min-width: 760px) {
    hr {
      margin: 20px 0;
      margin-top: 20px;

      &.first {
        margin-top: 0;
      }
    }
  }
}

