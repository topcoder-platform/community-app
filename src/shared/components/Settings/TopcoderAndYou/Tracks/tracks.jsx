
const tracks = {
  DESIGN: {
    id: 'design',
    name: 'Design',
    description: 'Website, mobile, and product design; UI and UX',
  },
  DEVELOP: {
    id: 'develop',
    name: 'Development',
    description: 'Software architecture, component assembly, application development, and bug hunting',
  },
  DATA_SCIENCE: {
    id: 'data_science',
    name: 'Data Science',
    description: 'Algorithms and data structures, statistical analysis',
  },
};

export default tracks;
