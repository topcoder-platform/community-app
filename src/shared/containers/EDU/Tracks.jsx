/**
 * Container for EDU Portal tracks page.
 */
import _ from 'lodash';
import moment from 'moment';
import React from 'react';
import { config, isomorphy } from 'topcoder-react-utils';
import Viewport from 'components/Contentful/Viewport';
import SearchBar from 'components/Contentful/SearchBar/SearchBar';
import { getService } from 'services/contentful';
import { tracksTreeBuilder } from 'utils/contentful';
import { updateQuery } from 'utils/url';
import qs from 'qs';
import TracksTree from 'components/Contentful/TracksTree/TracksTree';
import LoadingIndicator from 'components/LoadingIndicator';
import TracksFilter from 'components/Contentful/TracksFilter/TracksFilter';
// SVGs & Assets
import Dev from 'assets/images/img-development.png';
import Design from 'assets/images/img_design.png';
import DS from 'assets/images/img-data-science.png';
import Algo from 'assets/images/img-algorithm.png';
import QA from 'assets/images/img-QA.png';
import Topcoder from 'assets/images/img-Topcoder.png';
// Partials
import ResultTabs from './partials/ResultTabs';
// CSS
import tracksTheme from './styles/tracks.scss';

const TRACK_BANNER_BACK_COLORS = {
  Development: '#8AFB8A',
  Design: '#50ADE8',
  'Data Science': '#FFA45D',
  'Competitive Programming': '#FFA45D',
  QA: '#8AFB8A',
  Topcoder: '#2A2A2A',
};
const TRACK_IMAGES = {
  Development: Dev,
  Design,
  'Data Science': DS,
  'Competitive Programming': Algo,
  QA,
  Topcoder,
};

export default class EDUTracks extends React.Component {
  constructor(props) {
    super(props);
    // create a service to work with Contentful
    this.apiService = getService({ spaceName: 'EDU' });
    // init state
    this.state = {
      query: {
        tags: [],
      },
      tree: [],
      isShowFilter: false,
    };
    // bindings
    this.onTreeClick = this.onTreeClick.bind(this);
    this.onApplyFilter = this.onApplyFilter.bind(this);
  }

  componentDidMount() {
    // set current query
    let urlQuery = {};
    if (isomorphy.isClientSide()) {
      urlQuery = qs.parse(window.location.search.slice(1));
      // eslint-disable-next-line no-nested-ternary
      urlQuery.tags = _.isArray(urlQuery.tags)
        ? urlQuery.tags : (urlQuery.tags ? [urlQuery.tags] : []);
      this.setState({
        query: urlQuery,
      });
    }
    // Fire API requests
    // Get the EDU taxonomy
    this.apiService.getEDUTaxonomy()
      .then((taxonomy) => {
        const { query } = this.state;
        const tree = tracksTreeBuilder(taxonomy, query);
        this.setState({
          tree,
          taxonomy,
        });
      });
    // Get counters
    this.apiService.getEDUContent({
      track: urlQuery.track,
    })
      .then((content) => {
        this.setState({
          articleCnt: content.Article.total,
          videoCnt: content.Video.total,
          forumCnt: content['Forum post'].total,
        });
      });
  }

  /**
   * Tracks tree click on items
   * @param {Object} selectedItem
   */
  onTreeClick(selectedItem) {
    // Update the counters if track changed
    const { query } = this.state;
    // if track is not empty, it means clicked from child, otherwise clicked from parent
    const track = selectedItem.track ? selectedItem.track : selectedItem.title;
    const tax = selectedItem.track ? selectedItem.title : null;
    if (query.track !== track) {
      this.apiService.getEDUContent({
        track,
      })
        .then((content) => {
          this.setState({
            articleCnt: content.Article.total,
            videoCnt: content.Video.total,
            forumCnt: content['Forum post'].total,
          });
        });
    }
    // Update the state
    this.setState({
      query: _.merge(query, {
        track,
        tax,
      }),
    });
    // Update the url query
    updateQuery({
      track,
      tax,
    });
  }

  onApplyFilter(filterState) {
    const queryUpdate = {
      author: filterState.selectedAuthor,
      tags: filterState.tags,
      startDate: filterState.startDate.format(),
      endDate: filterState.endDate.format(),
    };
    // Update the state
    this.setState(prevState => ({
      query: {
        ...prevState.query,
        author: queryUpdate.author,
        tags: queryUpdate.tags,
        startDate: queryUpdate.startDate,
        endDate: queryUpdate.endDate,
      },
    }));
    // Update the url query
    updateQuery({ ...queryUpdate });
  }

  render() {
    const {
      taxonomy, query, tree, isShowFilter,
      articleCnt, videoCnt, forumCnt,
    } = this.state;
    // This container needs at least those variables
    // to be able to render meaningful data
    if (!taxonomy) return <LoadingIndicator />;
    return (
      <div className={tracksTheme.container}>
        {/* Banner */}
        <div
          className={tracksTheme.bannerContainer}
          style={{ backgroundColor: TRACK_BANNER_BACK_COLORS[query.track || 'Design'] }}
        >
          {
            TRACK_IMAGES[query.track] ? (
              <img src={TRACK_IMAGES[query.track]} alt="Track banner" className={tracksTheme.bannerImage} />
            ) : null
          }
          <div className={tracksTheme.bannerWrapp} style={query.track === 'Topcoder' ? { color: '#fff' } : {}}>
            <h1 className={tracksTheme.bannerText}>{query.track || 'THRIVE'}</h1>
            <div className={tracksTheme.bannerCounters}>
              <div className={tracksTheme.bannerCounterWrap}>
                <div className={tracksTheme.bannerCount}>{articleCnt}</div>
                <div className={tracksTheme.bannerCountLabel}>articles</div>
              </div>
              <div className={tracksTheme.bannerCounterWrap}>
                <div className={tracksTheme.bannerCount}>{videoCnt}</div>
                <div className={tracksTheme.bannerCountLabel}>videos</div>
              </div>
              <div className={tracksTheme.bannerCounterWrap}>
                <div className={tracksTheme.bannerCount}>{forumCnt}</div>
                <div className={tracksTheme.bannerCountLabel}>forum posts</div>
              </div>
            </div>
          </div>
          <div className={tracksTheme.searchBarWrapp}>
            <SearchBar />
          </div>
        </div>
        <div className={tracksTheme.shapeBanner} />
        {/* Main track content */}
        <div className={tracksTheme.main}>
          <div className={tracksTheme.tracksTree}>
            {
              tree.length ? (
                <TracksTree
                  list={tree}
                  onItemClick={this.onTreeClick}
                />
              ) : <LoadingIndicator />
            }
          </div>
          <div className={tracksTheme.tracksContent}>
            <div className={tracksTheme.tracksContentTop}>
              <h3 className={tracksTheme.trackTitle}>{query.tax || query.track}</h3>
              <button
                className={tracksTheme.filterToggle}
                type="button"
                onClick={() => { this.setState({ isShowFilter: !isShowFilter }); }}
              >
                {isShowFilter ? 'Hide filter' : 'Show filter'}
              </button>
            </div>
            <div className={isShowFilter ? tracksTheme.filterVisible : tracksTheme.filterHidden}>
              <TracksFilter
                onClose={() => this.setState({ isShowFilter: false })}
                onApply={this.onApplyFilter}
                selectedAuthor={query.author}
                tags={query.tags}
                startDate={query.startDate ? moment(query.startDate) : undefined}
                endDate={query.endDate ? moment(query.endDate) : undefined}
              />
            </div>
            <ResultTabs query={query} taxonomy={taxonomy} />
          </div>
        </div>
        {/* Top Design Videos & Recommended for you sections are Contentful editable via ID */}
        <Viewport
          id="CS6tscFxLUCZoPzWNB9Hv"
          spaceName="EDU"
          baseUrl={config.TC_EDU_BASE_PATH}
        />
      </div>
    );
  }
}
