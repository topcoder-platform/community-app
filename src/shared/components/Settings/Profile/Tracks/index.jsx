/**
 * Child component of Settings/Profile/Tracks renders "Tracks" section of profile setting page.
 */
/* global document */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';

import DesignIcon from 'assets/images/profile/ico-track-design.svg';
import DevelopIcon from 'assets/images/profile/ico-track-develop.svg';
import DataScienceIcon from 'assets/images/profile/ico-track-data.svg';

import DesignGreyIcon from 'assets/images/profile/ico-track-design-grey.svg';
import DevelopGreyIcon from 'assets/images/profile/ico-track-develop-grey.svg';
import DataScienceGreyIcon from 'assets/images/profile/ico-track-data-grey.svg';

import './styles.scss';

const tracks = {
  DESIGN: {
    label: 'Design',
    enabledIcon: DesignIcon,
    disabledIcon: DesignGreyIcon,
    description: 'Website, mobile, and product design; UI and UX',
  },
  DEVELOP: {
    label: 'Development',
    enabledIcon: DevelopIcon,
    disabledIcon: DevelopGreyIcon,
    description: 'Software architecture, component assembly, application development, and bug hunting',
  },
  DATA_SCIENCE: {
    label: 'Data Science',
    enabledIcon: DataScienceIcon,
    disabledIcon: DataScienceGreyIcon,
    description: 'Algorithms and data structures, statistical analysis',
  },
};

export default class Tracks extends React.Component {
  constructor(props) {
    super(props);
    this.onUpdateTracks = this.onUpdateTracks.bind(this);
  }

  onUpdateTracks() {
    const switches = document.querySelectorAll('input[name="onoffswitch"]');
    const newTracks = [];
    _.forEach(switches, (sw) => {
      if (sw.checked) {
        newTracks.push(sw.value);
      }
    });

    this.props.onUpdateTracks(newTracks);
  }

  render() {
    const userTracks = this.props.tracks;

    return (
      <div className="settings-section">
        <div className="section-info">
          <h2>Tracks</h2>
          <div className="description">Topcoder&apos;s three categories of challenges... please pick at least one based on your skills and interests.</div>
        </div>
        <div className="section-fields">
          <div styleName="track-toggle">
            {
              _.map(tracks, (track, trackName) => {
                const enabled = _.includes(userTracks, trackName);
                return (
                  <div styleName="track" key={trackName}>
                    <div styleName={enabled ? 'content' : 'content disabled'}>
                      <div styleName="track-details">
                        <div styleName={enabled ? 'icon' : 'icon disabled'}>
                          {
                            enabled &&
                            track.enabledIcon()
                          }
                          {
                            !enabled &&
                            track.disabledIcon()
                          }
                        </div>
                        <div styleName="text">
                          <span styleName={enabled ? 'title' : 'title disabled'}>{track.label}</span>
                          <div styleName="description">
                            <span>{track.description}</span>
                          </div>
                        </div>
                      </div>
                      <div className="onoffswitch">
                        <input
                          type="checkbox"
                          name="onoffswitch"
                          id={`DESIGN-onoffswitch-${trackName}`}
                          value={trackName}
                          defaultChecked={enabled}
                          onChange={this.onUpdateTracks}
                          className="onoffswitch-checkbox"
                        />
                        <label htmlFor={`DESIGN-onoffswitch-${trackName}`} className="onoffswitch-label">
                          <span className="onoffswitch-inner" />
                          <span className="onoffswitch-switch" />
                        </label>
                      </div>
                    </div>
                  </div>
                );
              })
            }
          </div>
        </div>
      </div>
    );
  }
}

Tracks.defaultProps = {
  tracks: [],
};

Tracks.propTypes = {
  tracks: PT.arrayOf(PT.string),
  onUpdateTracks: PT.func.isRequired,
};

