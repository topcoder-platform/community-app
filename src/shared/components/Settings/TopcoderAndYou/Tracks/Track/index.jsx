/**
 * Render Track Component
 */
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import { isomorphy } from 'topcoder-react-utils';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/profile/tracks', false, /svg/);
}

export default function Track({
  checked,
  id,
  onToggle,
  title,
  description,
  value,
}) {
  const icon = `./${id.toLowerCase()}.svg`;
  return (
    <div styleName="track-container">
      <div styleName="body">
        <div styleName="icon">
          { assets && assets.keys().includes(icon) ? <ReactSVG path={assets(`${icon}`)} /> : '' }
        </div>
        <div styleName="info">
          <div styleName="info-first-line">
            { title }
          </div>
          <div styleName="info-second-line">
            { description }
          </div>
        </div>
      </div>
      <div styleName="onoffswitch-no-padding-right">
        <input
          type="checkbox"
          name="eprf-onoffswitch"
          id={`pre-onoffswitch-${id}`}
          value={value}
          checked={checked}
          onChange={onToggle}
          className="onoffswitch-checkbox"
        />
        <label htmlFor={`pre-onoffswitch-${id}`} className="onoffswitch-label">
          <span className="onoffswitch-inner" />
          <span className="onoffswitch-switch" />
          <input type="hidden" />
        </label>
      </div>
    </div>
  );
}

Track.propTypes = {
  id: PT.string.isRequired,
  value: PT.string.isRequired,
  checked: PT.bool.isRequired,
  title: PT.string.isRequired,
  description: PT.string.isRequired,
  onToggle: PT.func.isRequired,
};
