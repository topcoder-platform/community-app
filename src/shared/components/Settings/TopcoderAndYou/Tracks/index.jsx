/**
 * Tracks
 * Topcoder & You Settings Page
 */

import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import tracks from './tracks';
import Track from './Track';

import '../styles.scss';

const Tracks = (props) => {
  const {
    newProfileInfo,
    onChange,
  } = props;

  return (
    <div styleName="you-and-topcoder-container">
      <div styleName="track-list">
        {
          _.map(tracks, (track) => {
            const result = newProfileInfo.tracks.filter(item => (
              item.toUpperCase() === track.id.toUpperCase()
            ));
            const checked = result.length !== 0;
            return (
              <Track
                icon={track.icon}
                key={track.id}
                id={track.id}
                value={track.id}
                checked={checked}
                title={track.name}
                description={track.description}
                onToggle={event => onChange(track.id, event.target.checked)}
              />
            );
          })
        }
      </div>
    </div>
  );
};

Tracks.defaultProps = {
  newProfileInfo: {},
};

Tracks.propTypes = {
  newProfileInfo: PT.shape(),
  onChange: PT.func.isRequired,
};

export default Tracks;
