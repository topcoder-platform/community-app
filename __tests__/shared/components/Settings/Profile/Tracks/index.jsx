import React from 'react';
import Renderer from 'react-test-renderer/shallow';

import Tracks from 'components/Settings/Profile/Tracks';

import userProfile from '../../__mocks__/user-profile.json';

const rnd = new Renderer();

it('renders tracks section of profile setting page correctly', () => {
  rnd.render((<Tracks
    tracks={userProfile.tracks}
    onUpdateTracks={() => { }}
  />));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
