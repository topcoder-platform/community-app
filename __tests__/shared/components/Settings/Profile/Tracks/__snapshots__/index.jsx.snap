// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tracks section of profile setting page correctly 1`] = `
<div
  className="settings-section"
>
  <div
    className="section-info"
  >
    <h2>
      Tracks
    </h2>
    <div
      className="description"
    >
      Topcoder's three categories of challenges... please pick at least one based on your skills and interests.
    </div>
  </div>
  <div
    className="section-fields"
  >
    <div
      className="src-shared-components-Settings-Profile-Tracks-___styles__track-toggle___OB_8P"
    >
      <div
        className="src-shared-components-Settings-Profile-Tracks-___styles__track___2c2sB"
      >
        <div
          className="src-shared-components-Settings-Profile-Tracks-___styles__content___2vWqc src-shared-components-Settings-Profile-Tracks-___styles__disabled___1rvBf"
        >
          <div
            className="src-shared-components-Settings-Profile-Tracks-___styles__track-details___3lh6X"
          >
            <div
              className="src-shared-components-Settings-Profile-Tracks-___styles__icon___CfGY1 src-shared-components-Settings-Profile-Tracks-___styles__disabled___1rvBf"
            >
              <svg>
                <title>
                  ico-track-design-grey
                </title>
                <g
                  fill="none"
                  fillRule="evenodd"
                  stroke="#A3A3AE"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="1.2"
                >
                  <path
                    d="M11.045 17.327L24.823 3.602a3.776 3.776 0 1 1 5.34 5.34L16.386 22.669l-5.34-5.34zM10.686 23.195c-1.621-1.62-4.237-1.633-5.842-.027-1.026 1.025-1.164 2.298-1.152 3.363.012 1.052-.74 2.014-2.578 2.045 2.581 2.58 7.077 2.985 9.6.462 1.7-1.7 1.45-4.365-.028-5.843zM12.608 5.204L9.404 8.41M27.562 20.158l-3.205 3.204M19.135 26.685a1735309.675 1735309.675 0 0 1 4.154 4.154l7.477-7.477s-4.544-4.862-4.751-5.078M14.55 7.134L9.405 2 1.927 9.477l4.034 4.034"
                  />
                </g>
              </svg>
            </div>
            <div
              className="src-shared-components-Settings-Profile-Tracks-___styles__text___2txAw"
            >
              <span
                className="src-shared-components-Settings-Profile-Tracks-___styles__title___2dRZk src-shared-components-Settings-Profile-Tracks-___styles__disabled___1rvBf"
              >
                Design
              </span>
              <div
                className="src-shared-components-Settings-Profile-Tracks-___styles__description___1TAW2"
              >
                <span>
                  Website, mobile, and product design; UI and UX
                </span>
              </div>
            </div>
          </div>
          <div
            className="onoffswitch"
          >
            <input
              className="onoffswitch-checkbox"
              defaultChecked={false}
              id="DESIGN-onoffswitch-DESIGN"
              name="onoffswitch"
              onChange={[Function]}
              type="checkbox"
              value="DESIGN"
            />
            <label
              className="onoffswitch-label"
              htmlFor="DESIGN-onoffswitch-DESIGN"
            >
              <span
                className="onoffswitch-inner"
              />
              <span
                className="onoffswitch-switch"
              />
            </label>
          </div>
        </div>
      </div>
      <div
        className="src-shared-components-Settings-Profile-Tracks-___styles__track___2c2sB"
      >
        <div
          className="src-shared-components-Settings-Profile-Tracks-___styles__content___2vWqc"
        >
          <div
            className="src-shared-components-Settings-Profile-Tracks-___styles__track-details___3lh6X"
          >
            <div
              className="src-shared-components-Settings-Profile-Tracks-___styles__icon___CfGY1"
            >
              <svg>
                <title>
                  ico-track-develop
                </title>
                <g
                  fill="none"
                  fillRule="evenodd"
                  stroke="#05C14F"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="1.2"
                >
                  <path
                    d="M13.5 30.5h-11v-8h11v8zM13.141 6.065h10.355v3.27M14.35 26.893h9.209v-4.231M12.714 6.357a5.357 5.357 0 1 1-10.713 0 5.357 5.357 0 0 1 10.713 0z"
                  />
                  <path
                    d="M23.526 9.123l-6.867 6.867 6.867 6.867 6.867-6.867-6.867-6.867z"
                  />
                </g>
              </svg>
            </div>
            <div
              className="src-shared-components-Settings-Profile-Tracks-___styles__text___2txAw"
            >
              <span
                className="src-shared-components-Settings-Profile-Tracks-___styles__title___2dRZk"
              >
                Development
              </span>
              <div
                className="src-shared-components-Settings-Profile-Tracks-___styles__description___1TAW2"
              >
                <span>
                  Software architecture, component assembly, application development, and bug hunting
                </span>
              </div>
            </div>
          </div>
          <div
            className="onoffswitch"
          >
            <input
              className="onoffswitch-checkbox"
              defaultChecked={true}
              id="DESIGN-onoffswitch-DEVELOP"
              name="onoffswitch"
              onChange={[Function]}
              type="checkbox"
              value="DEVELOP"
            />
            <label
              className="onoffswitch-label"
              htmlFor="DESIGN-onoffswitch-DEVELOP"
            >
              <span
                className="onoffswitch-inner"
              />
              <span
                className="onoffswitch-switch"
              />
            </label>
          </div>
        </div>
      </div>
      <div
        className="src-shared-components-Settings-Profile-Tracks-___styles__track___2c2sB"
      >
        <div
          className="src-shared-components-Settings-Profile-Tracks-___styles__content___2vWqc src-shared-components-Settings-Profile-Tracks-___styles__disabled___1rvBf"
        >
          <div
            className="src-shared-components-Settings-Profile-Tracks-___styles__track-details___3lh6X"
          >
            <div
              className="src-shared-components-Settings-Profile-Tracks-___styles__icon___CfGY1 src-shared-components-Settings-Profile-Tracks-___styles__disabled___1rvBf"
            >
              <svg>
                <title>
                  ico-track-data-grey
                </title>
                <g
                  fill="none"
                  fillRule="evenodd"
                  stroke="#A3A3AE"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="1.2"
                >
                  <path
                    d="M9.488 3.857C9.488 2.28 10.755 1 12.318 1c1.562 0 2.828 1.28 2.828 2.857 0 1.578-1.266 2.857-2.829 2.857-1.562 0-2.83-1.279-2.83-2.857zM24.341 9.571c0-1.577 1.267-2.857 2.83-2.857 1.562 0 2.829 1.28 2.829 2.857 0 1.578-1.267 2.858-2.83 2.858-1.562 0-2.829-1.28-2.829-2.858zM1 19.215c0-1.578 1.267-2.857 2.83-2.857 1.562 0 2.829 1.279 2.829 2.857 0 1.578-1.267 2.857-2.83 2.857-1.562 0-2.829-1.28-2.829-2.857zM13.024 27.786c0-1.578 1.267-2.857 2.83-2.857 1.562 0 2.829 1.28 2.829 2.857 0 1.578-1.267 2.857-2.83 2.857-1.562 0-2.829-1.279-2.829-2.857zM24.341 22.072c0-1.578 1.267-2.857 2.83-2.857 1.562 0 2.829 1.28 2.829 2.857 0 1.578-1.267 2.857-2.83 2.857-1.562 0-2.829-1.28-2.829-2.857zM5.598 16.268l5.481-9.643M27.082 19.125v-6.429M15.677 24.839l-2.83-17.768M25.049 20.017L14.35 6.267M18.771 26.803l5.836-3.303M6.924 18.232L24.607 11"
                  />
                </g>
              </svg>
            </div>
            <div
              className="src-shared-components-Settings-Profile-Tracks-___styles__text___2txAw"
            >
              <span
                className="src-shared-components-Settings-Profile-Tracks-___styles__title___2dRZk src-shared-components-Settings-Profile-Tracks-___styles__disabled___1rvBf"
              >
                Data Science
              </span>
              <div
                className="src-shared-components-Settings-Profile-Tracks-___styles__description___1TAW2"
              >
                <span>
                  Algorithms and data structures, statistical analysis
                </span>
              </div>
            </div>
          </div>
          <div
            className="onoffswitch"
          >
            <input
              className="onoffswitch-checkbox"
              defaultChecked={false}
              id="DESIGN-onoffswitch-DATA_SCIENCE"
              name="onoffswitch"
              onChange={[Function]}
              type="checkbox"
              value="DATA_SCIENCE"
            />
            <label
              className="onoffswitch-label"
              htmlFor="DESIGN-onoffswitch-DATA_SCIENCE"
            >
              <span
                className="onoffswitch-inner"
              />
              <span
                className="onoffswitch-switch"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
